prbt_hardware_support/include/prbt_hardware_support/service_client_factory.h

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2019 Pilz GmbH & Co. KG
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef PRBT_HARDWARE_SUPPORT_SERVICE_CLIENT_FACTORY_H
+#define PRBT_HARDWARE_SUPPORT_SERVICE_CLIENT_FACTORY_H
+
+#include <string>
+
+#include <ros/service.h>
+#include <prbt_hardware_support/wait_for_service.h>
+
+namespace prbt_hardware_support
+{
+
+class ServiceClientFactory
+{
+public:
+  /**
+   * @brief Waits for a service to exist and creates a ServiceClient.
+   *
+   * @param name Name of the service, to be resolved in the node namespace
+   * @return The ServiceClient
+   */
+  template <typename Service>
+  inline static ros::ServiceClient create(const std::string &name)
+  {
+    waitForService(name);
```

Comment:
better add `while(ros::ok) { ROS_INFO_THROTTLED...}` as in http://wiki.ros.org/roscpp/Overview/Logging 
see https://github.com/PilzDE/pilz_robots/blob/10b4fb75ee28a35107087a024a174cd67026305f/prbt_hardware_support/include/prbt_hardware_support/wait_for_service.h#L38-L43

Reply:
this is exactly the waitForService() that is called ...

Reply:
okay. To avoid further confusion I'd suggest to unify both. Either keep your Factory class and refactor  braketest accordingly or stick to Hagen's functional approach and remove your class. Why would you need two approaches (files, tests, documentation) for the same functionality?
https://github.com/PilzDE/pilz_robots/blob/8bd4e2042462b7dc2768748bded6e2e150d234cb/prbt_hardware_support/src/brake_test_executor.cpp#L49-L58

Reply:
I agree. I would suggest to do the refactoring in a separate PR and concentrate on the STO fix here.

Reply:
Also I think the confusion is between ros::service::waitForService() and prbt_hardware_support::waitForService(), which can be eliminated by refactoring.